app/controllers/bookings_controller.rb

```diff
@@ -0,0 +1,87 @@
+class BookingsController < ApplicationController
+  before_action :set_booking, only: [:show, :edit, :update, :destroy]
+
+  # GET /bookings
+  # GET /bookings.json
+  def index
+    @bookings = Booking.all
+  end
+
+  # GET /bookings/1
+  # GET /bookings/1.json
+  def show
+  end
+
+  # GET /bookings/new
+  def new
+    @booking = Booking.new(:car_id => params[:car_id])
+    @booking.build_customer
+    @booking.references.build
+  end
+
+  # GET /bookings/1/edit
+  def edit
+  end
+
+  # POST /bookings
+  # POST /bookings.json
+  def create
+    @booking = Booking.new(booking_params)
+
+    respond_to do |format|
+      if @booking.save
+        format.html { redirect_to @booking, notice: 'Booking was successfully created.' }
+        format.json { render :show, status: :created, location: @booking }
+      else
+        format.html { render :new }
+        format.json { render json: @booking.errors, status: :unprocessable_entity }
+      end
+        # update number of days  and booking payment in booking table
+      # @booking.update_attributes({:number_of_days => (@booking.to_booking.to_date - @booking.from_booking.to_date).to_i, :booking_price=> (@booking.car.per_day_rent * @booking.number_of_days).to_i})
+
+      # update number of days in booking table
+
+      @booking.update_attribute(:number_of_days,(@booking.to_booking.to_date - @booking.from_booking.to_date).to_i)
```

Comment:
- make text field for number of days
- Calculate number of days at user end.
- Set value in number of days
- permit this attribute in booking_params